salt/utils/templates.py

```diff
@@ -345,11 +346,17 @@ def render_jinja_tmpl(tmplstr, context, tmplpath=None):
                                        **env_args)
 
     jinja_env.filters['strftime'] = salt.utils.date_format
-    jinja_env.filters['sequence'] = salt.utils.jinja.ensure_sequence_filter
     jinja_env.filters['yaml_dquote'] = salt.utils.yamlencoding.yaml_dquote
     jinja_env.filters['yaml_squote'] = salt.utils.yamlencoding.yaml_squote
     jinja_env.filters['yaml_encode'] = salt.utils.yamlencoding.yaml_encode
 
+    for function_name in context['salt'].keys():
+        # Simply iterating over the keys forces the execution modules to load
```

Comment:
Okay,  so now it works also for execution modules thanks to that iteration. I don't see any for the utils. But I can see that you are doing the same mistake as previously, rewriting the same object. I would be surprised if that would work with more than one function from the utils!

> The benefit to this implementation is that we can use the same approach to adding jinja filters across the salt code base, rather than having one approach for execution modules and another approach for utils modules.

I agree, that would be nice to have. At the same time, bear in mind that you create yet another object for each function having the decorator, as a new instance of your class. I think the performance of this is even more poor. Whilst the other one simply updates two properties of the existing objects.